platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/InboundConnectionHandler.java

```diff
@@ -50,83 +48,57 @@
-    private final AddressBook addressBook;
     private final InterruptableConsumer<Connection> newConnectionConsumer;
     private final SocketConfig socketConfig;
-    private final boolean doVersionCheck;
-    private final SoftwareVersion softwareVersion;
     /** Rate Limited Logger for SocketExceptions */
     private final RateLimitedLogger socketExceptionLogger;
 
     private final PlatformContext platformContext;
+    private final NetworkPeerIdentifier networkPeerIdentifier;
 
     public InboundConnectionHandler(
             @NonNull final PlatformContext platformContext,
             @NonNull final ConnectionTracker connectionTracker,
+            @NonNull final NetworkPeerIdentifier networkPeerIdentifier,
             @NonNull final NodeId selfId,
-            @NonNull final AddressBook addressBook,
             @NonNull final InterruptableConsumer<Connection> newConnectionConsumer,
-            final boolean doVersionCheck,
-            @NonNull final SoftwareVersion softwareVersion,
             @NonNull final Time time) {
         this.platformContext = Objects.requireNonNull(platformContext);
         this.connectionTracker = Objects.requireNonNull(connectionTracker);
         this.selfId = Objects.requireNonNull(selfId);
-        this.addressBook = Objects.requireNonNull(addressBook);
         this.newConnectionConsumer = Objects.requireNonNull(newConnectionConsumer);
-        this.doVersionCheck = doVersionCheck;
-        this.softwareVersion = Objects.requireNonNull(softwareVersion);
         Objects.requireNonNull(time);
         this.socketExceptionLogger = new RateLimitedLogger(logger, time, Duration.ofMinutes(1));
         this.socketConfig = platformContext.getConfiguration().getConfigData(SocketConfig.class);
+        this.networkPeerIdentifier = networkPeerIdentifier;
     }
 
     /**
-     * Authenticate the peer that has just established a new connection and create a {@link Connection}
+     * Identifies the peer that has just established a new connection and create a {@link Connection}
      *
      * @param clientSocket the newly created socket
      */
-    public void handle(final Socket clientSocket) {
-        SerializableDataInputStream dis = null;
-        SerializableDataOutputStream dos = null;
-        NodeId otherId = null;
+    public void handle(final Socket clientSocket, final List<PeerInfo> peerInfoList) {
         long acceptTime = 0;
+        NodeId otherId = null;
         try {
             acceptTime = System.currentTimeMillis();
             clientSocket.setTcpNoDelay(socketConfig.tcpNoDelay());
             clientSocket.setSoTimeout(socketConfig.timeoutSyncClientSocket());
 
+            if (clientSocket instanceof final SSLSocket sslSocket) {
+                final PeerInfo connectedPeer = networkPeerIdentifier.identifyTlsPeer(
+                        sslSocket.getSession().getPeerCertificates(), peerInfoList);
+                otherId = Objects.requireNonNull(connectedPeer).nodeId();
```

Comment:
identifyTlsPeer() returns null if it can't identify the peer, which means an unidentified peer will cause this this to throw a nullpointer exception. I'm wondering if it might be more appropriate to throw a more specific exception.

Also, we want to be careful to ensure that this failure will not generate excessive amounts of log spam (throwing the proper exception type plays into this). If we can't identify a peer, the amount of logging we do should be no more than if the connection breaks for any other reason.

Reply:
I think the logic here is that if we can't identify a peer, it's just regular business ... we log it and close the socket. The logging currently happens in NetworkPeerIdentifier#identifyTlsPeer, and the closing happens here when we throw the NPE. I agree it can be better though so I've changed it to:
`
            final SSLSocket sslSocket = ((SSLSocket) clientSocket);
            final PeerInfo connectedPeer = networkPeerIdentifier.identifyTlsPeer(
                    sslSocket.getSession().getPeerCertificates(), peerInfoList);

            if (connectedPeer == null) {
                sslSocket.close();
                return;
            }
            otherId = connectedPeer.nodeId();
`
This gets rid of the instanceOf check and fails if we can't cast (we won't because it's always an ssl socket), and also handles the null check more clearly.
(We are getting rid of Plain tcp soon, so the argument to this method will change to SSLSocket instead of Socket)